core/src/main/scala/org/apache/gearpump/cluster/client/ClientContext.scala

```diff
@@ -69,15 +80,14 @@ class ClientContext(masters: Iterable[HostPort]) {
 
   private def loadFile(jarPath : String) : AppJar = {
     val jarFile = new java.io.File(jarPath)
-    val fis = new FileInputStream(jarFile)
-    val bos: ByteArrayOutputStream = new ByteArrayOutputStream()
-    val buf = ListBuffer[Byte]()
-    var b = fis.read()
-    while (b != -1) {
-      buf.append(b.byteValue)
-      b = fis.read()
+
+    val uploadFile = (master ? GetJarFileContainer).asInstanceOf[Future[JarFileContainer]]
+      .map {container =>
+      container.copyFromLocal(new java.io.File(jarPath))
+      AppJar(jarFile.getName, container)
```

Comment:
It will be renamed in JarStore. On File Sever, it will pick a random name.

On executor, it will use name: $jarName +  random String